app/controllers/twilio_controller.rb

```diff
@@ -1,14 +1,41 @@
 class TwilioController < ApplicationController
   skip_before_action :verify_authenticity_token
 
+  def receive_text
+    recieve_phone(true, params)
+  end
+
   def receive_call
-    # TODO implement call_handler
+    recieve_phone(false, params)
   end
 
-  def receive_text
-    phone = params[:From].to_s
-    user = User.where(["phone_number = ?", phone]).first
-    response = TextHandler.new.process_input(user, params)
-    render xml: TwilioHandler.new.send_respond_text(response).to_s
+  private
+
+  def recieve_phone(is_text, params)
```

Comment:
Small typo here: method should be `receive_phone` instead of `recieve_phone` (receive is misspelled, may cause an error down the line if we try to call it later)